lib/engine-addon.js

```diff
@@ -488,8 +488,9 @@ module.exports = {
       };
 
       this.treeForEngine = function() {
-        // If this engine is lazy or any of its parents are lazy we need to promote its routes.
-        if (!this._hasLazyAncestor) { return; }
+        // Unless includeRoutesInApplication is false, if this engine or any of
+        // its parents are lazy we need to promote its routes.
+        if (!this._hasLazyAncestor || this.lazyLoading.includeRoutesInApplication === false) { return; }
```

Comment:
Added a block comment and a descriptive variable name to try and make this function a bit easier to understand.